src/transformers/models/m2m_100/modeling_m2m_100.py

```diff
@@ -1491,3 +1492,284 @@ def _reorder_cache(past_key_values, beam_idx):
                 tuple(past_state.index_select(0, beam_idx.to(past_state.device)) for past_state in layer_past),
             )
         return reordered_past
+
+
+@add_start_docstrings(
+    "The M2M100 (SONAR) transformer decoder model for generating texts from embeddings.",
+    M2M_100_START_DOCSTRING,
+)
+class M2M100DecoderModel(M2M100PreTrainedModel):
```

Comment:
this one does make sense as you have the lm_head

Reply:
Exactly!